src/targets/maven.ts

```diff
@@ -355,14 +391,33 @@ export class MavenTarget extends BaseTarget {
    * @param distDir directory of the distribution.
    * @returns record of required files.
    */
-  private getFilesForMavenPomDist(distDir: string): Record<string, string> {
+  private getFilesForMavenPomDist(distDir: string): Record<string, string | string[]> {
     const moduleName = parse(distDir).base;
-    return {
+    const files = {
       targetFile: join(distDir, this.getTargetFilename(distDir)),
       javadocFile: join(distDir, `${moduleName}-javadoc.jar`),
       sourcesFile: join(distDir, `${moduleName}-sources.jar`),
       pomFile: join(distDir, 'pom-default.xml'),
     };
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}-all.jar`))) {
+      Object.assign(files, { allFile: join(distDir, `${moduleName}-all.jar`)})
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}.module`))) {
+      Object.assign(files, { moduleFile: join(distDir, `${moduleName}.module`) })
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}-metadata.jar`))) {
+      Object.assign(files, { metadataFile: join(distDir, `${moduleName}-metadata.jar`) })
+    }
+    if (existsSync(join(distDir, `${moduleName.toLowerCase()}.klib`))) {
+      const cinteropFiles: string[] = []
+      readdirSync(distDir).forEach(file => {
```

Comment:
This code is potentially running for every artifact. Can we read the directory async? You can do this with `fsPromises.readdir`, and see an [example on line 281](https://github.com/getsentry/craft/pull/412/files#diff-d138225e2b0f12b719ddf3deee29e05257b82ef40f9adf1c7aa880696e7237f1R281).